File: Project/FileReportExporter.cpp
#include "FileReportExporter.hpp"

FileReportExporter::FileReportExporter(const std::string& filePath, const std::string& title, const std::string& body, bool includeDate){
	m_filePath = filePath;
	m_title = title;
	m_body = body;
	m_includeDate = includeDate;
}

FileReportExporter::FileReportExporter(const FileReportExporter& other) {
	m_filePath = other.m_filePath;
	m_title = other.m_title;
	m_body = other.m_body;
	m_includeDate = other.m_includeDate;
}

std::string FileReportExporter::GetFilePath() const {
	return m_filePath;
}

void FileReportExporter::SetFilePath(const std::string& value) {
	m_filePath = value;
}

std::string FileReportExporter::GetTitle() const {
	return m_title;
}

void FileReportExporter::SetTitle(const std::string& value) {
	m_title = value;
}

std::string FileReportExporter::GetBody() const {
	return m_body;
}

void FileReportExporter::SetBody(const std::string& value) {
	m_body = value;
}

bool FileReportExporter::GetIncludeDate() const {
	return m_includeDate;
}

void FileReportExporter::SetIncludeDate(bool value) {
	m_includeDate = value;
}

File: ProjectCS/GroupTable/FileReportExporter.cs
using System;
using System.Collections.Generic;

namespace SessionResultsCS
{
    public abstract class FileReportExporter : IReportExporter
    {
        // Путь к файлу для сохранения отчёта
        public string FilePath { get; set; }
        // Заголовок отчета
        public string Title { get; set; }
        // Текст отчета
        public string Body { get; set; }
        // Включать ли дату генерации в отчет
        public bool IncludeDate { get; set; }


        public FileReportExporter(string filePath, string title = "", string body = "", bool includeDate = false)
        {
            FilePath = filePath;
            Title = title;
            Body = body;
            IncludeDate = includeDate;
        }


        // Экспортирует таблицу группы
        public abstract void Export(GroupTable groupTable);
    }
}
